pilot/pkg/model/envoyfilter.go

```diff
@@ -75,10 +78,35 @@ var wellKnownVersions = map[string]string{
 func convertToEnvoyFilterWrapper(local *config.Config) *EnvoyFilterWrapper {
 	localEnvoyFilter := local.Spec.(*networking.EnvoyFilter)
 
-	out := &EnvoyFilterWrapper{Name: local.Name, Namespace: local.Namespace, Priority: localEnvoyFilter.Priority, creationTime: local.CreationTimestamp}
+	out := &EnvoyFilterWrapper{
+		Name:                         local.Name,
+		Namespace:                    local.Namespace,
+		ReferencedNamespacedServices: nil,
+		ReferencedServices:           nil,
+		workloadSelector:             nil,
+		Patches:                      nil,
+		Priority:                     localEnvoyFilter.Priority,
+		creationTime:                 local.CreationTimestamp,
+	}
 	if localEnvoyFilter.WorkloadSelector != nil {
 		out.workloadSelector = localEnvoyFilter.WorkloadSelector.Labels
 	}
+
+	svcRefs := local.Annotations["envoyfilter.istio.io/referenced-services"]
```

Comment:
is n't this now inconsistent with sidecars? We now only process this annotation for gateways and waypoints but ignore for sidecars?

Reply:
For sidecars there is no need to handle this so not sure it matters?

Reply:
I guess unless we want to explicitly allow it even if its excluded by Sidecar? In my mind it works like this:

For gateway (with filter)...:
* Start with no services
* Add all referenced by VS
* Add all referenced by extensionProvider
* Add all referenced by Envoyfilter

For sidecar:
* Start with all services
* Add all referenced by EnvoyFilter (NO-OP, we added them already)
* Filter with Sidecar

Reply:
do you plan to document this annotation?

Reply:
Yeah I can add it to the docs if we merge this